### PR TITLE
HPCC-29553 Ensure bare-metal dropzones create groups as well as planes

### PR DESCRIPTION
**Pull request by jakesmith**
Ensure groups exist (in Dali) for dropzone planes. Also, push plane/group names in spray/despray requests in ws_fsService, through to DFU job source/target specifications.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by ghalliday — COMMENTED** [review]
@jakesmith looks reasonable.  One question (which I expect the answer to be no).  Main concerns would be any backward compatibility issues.  Also, the PR for the linked jira is currently targetting 9.0.x (#17292).  Should that change?

**Comment by jakesmith** [comment]
> @jakesmith looks reasonable. One question (which I expect the answer to be no). Main concerns would be any backward compatibility issues.

I don't think so.
New bare-metal builds with this change, will build out dropzone groups that will not have been there (in Dali) before, I think that in itself is harmless, matching clients should build out planes based on those groups, if they were for some reason old clients, they would build planes without groups based on environment as they did before - that would not pose an incompatibility issue.
If Dali was behind and clients ahead - then.. they would not build dropzone planes, not sure that is a big concern for 9.2


>Also, the PR for the linked jira is currently targetting 9.0.x (#17292). Should that change?

yes, I've commented there.

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -10462,6 +10481,27 @@ class CInitGroups
...
+        Owned<IPropertyTreeIterator> dropzones = conn->queryRoot()->getElements("DropZone");
+        ForEach(*dropzones)
+        {
+            IPropertyTree & cur = dropzones->query();
+            unsigned numServers = cur.getCount("ServerList");
+
+            //Allow url style drop zones, and drop zones with a single node.  Not sure what >1 would mean in legacy.
+            if (numServers <= 1)
+            {
+                IPropertyTree &dropZone = dropzones->query();
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dadfs.cpp:10495]: will change.

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -10153,6 +10167,11 @@ class CInitGroups
             case grp_roxie:
                 gname.append(cluster.queryProp("@name"));
                 break;
+            case grp_dropzone:
+                gname.append(cluster.queryProp("@name"));
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dadfs.cpp:10171]: 99% sure not, the xslts reference name, the stock has name, I think we depend on it elsewhere.
Could guard against it here, but really don't think should ever be missing.

File: dali/base/dadfs.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dadfs.cpp]: @ghalliday - pushed new commit with change.

**Comment by jakesmith** [comment]
> @ghalliday - regression test 'spray_dir_test.xml' in roxie failed. I'll investigate.

@ghalliday - ready for review

**Review by ghalliday — APPROVED** [review]
@jakesmith please squash

**Comment by jakesmith** [comment]
> @jakesmith please squash

@ghalliday - done.

**Pull request merged**